main/zoe/guide/README.md

```diff
@@ -64,176 +63,170 @@ handle the credit cards themselves.
-    customProperties: {
-      asset: give.Asset,
-      price: want.Price,
-      inviteDesc: 'matchOffer',
-    },
-  });
-};
+{ give: { Asset: amount1 }, want: { Price: amount2 } }
 ```
-In the `makeMatchingInvite` method we call the [`swap` helper function](../api/zoe-helpers.html#zoehelper-swap-keephandle-tryhandle-keephandleinactivemsg), which handles a lot of the logic. The code for `swap` is:
+`amount1` and `amount2` are amounts with the correct issuers for the keywords.
+`assertProposalShape()` then pulls out the elements **tyg todo: Which are the elements? The amounts?** of the 
+proposal so it can later match them.
 
 ```js
-  swap: (
-    keepHandle,
-    tryHandle,
-    keepHandleInactiveMsg = 'prior offer is unavailable',
-  ) => {
-    if (!zcf.isOfferActive(keepHandle)) {
-      throw helpers.rejectOffer(tryHandle, keepHandleInactiveMsg);
-    }
-    if (!helpers.canTradeWith(keepHandle, tryHandle)) {
-      throw helpers.rejectOffer(tryHandle);
-    }
-    const keepAmounts = zcf.getCurrentAllocation(keepHandle);
-    const tryAmounts = zcf.getCurrentAllocation(tryHandle);
-    // reallocate by switching the amount
-    const handles =         harden([keepHandle, tryHandle]);
-    zcf.reallocate(handles, harden([tryAmounts, keepAmounts]));
-    zcf.complete(handles);
-    return defaultAcceptanceMsg;
-  }
+  const makeMatchingInvitation = firstSeat => {
+    assertProposalShape(firstSeat, {
+      give: { Asset: null },
+      want: { Price: null },
+    });
+    const { want, give } = firstSeat.getProposal();
 ```
-First, `swap` checks if the offer is still 
-active. If not, we reject the offer at hand. Second, if the offer 
-at hand isn't a match for the first offer, we want to reject it 
-for that reason as well.
 
-Once we're sure that we *do* have a matching offer, we can do the most
-exciting part, the reallocation.
+`makeMatchingInvitation()`, our first handler, then constructs a handler for the second offer,
+with the first offer's `want` and `give` in scope. This second
+handler, `matchingSeatOfferHandler()` does the final step.
+It uses the [`swap` helper  function](../api/zoe-helpers.html#swap-zcf-leftseat-rightseat-lefthasexitedmsg-righthasexitedmsg),
+a powerful Zoe Helper that handles a lot of the logic of doing a basic swap of assets.
 
+When the swap succeeds, it exits both seats and the contract shuts down. **tyg todo: Should we mention reallocation of assets between the parties?**
+```js
+    const matchingSeatOfferHandler = matchingSeat => {
+      const swapResult = swap(zcf, firstSeat, matchingSeat);
+      zcf.shutdown();
+      return swapResult;
+    };
+```
+Now let's put it together. The last step of the first handler, `makeMatchingInvitation()`
+is to create and return the second party's invitation, using
+`matchingSeatOfferHandler()` and including custom properties for the expected proposal.  **tyg todo: Maybe "offer proposal" to reinforce the connection?**
```

Comment:
`matchingSeatOfferHandler` and including custom properties for the proposal of the invited offer.